tests/robustness/model/history.go

```diff
@@ -300,10 +301,11 @@ func (h *AppendableHistory) appendFailed(request EtcdRequest, start, end time.Du
 }
 
 func (h *AppendableHistory) append(op porcupine.Operation) {
-	if op.Return != -1 && op.Call >= op.Return {
+	if op.Return != math.MaxInt64 && op.Call >= op.Return {
 		panic(fmt.Sprintf("Invalid operation, call(%d) >= return(%d)", op.Call, op.Return))
 	}
-	if len(h.operations) > 0 {
+
+	if len(h.operations) > 0 && op.ClientId == h.operations[len(h.operations)-1].ClientId {
```

Comment:
I think in the end this conditional check is enough, instead of the need of using the map as we discussed offline, because the only place that the `ClientId` would be incremented is in `appendFailed` (code snippet attached below)

```
func (h *AppendableHistory) appendFailed(request EtcdRequest, start, end time.Duration, err error) {
	op := porcupine.Operation{
		ClientId: h.streamID, /* ClientId is set here */
		Input:    request,
		Call:     start.Nanoseconds(),
		Output:   failedResponse(err),
		Return:   end.Nanoseconds(),
	}
	isRead := request.IsRead()
	if !isRead {
		op.Return = math.MaxInt64
		h.streamID = h.idProvider.NewStreamID() /* ClientId is incremented here only in the entire test */
	}
	h.append(op)
}
```

Reply:
I will do it in a follow-up as I have a half-done PR already! 

I am also going to add clientID to the unit test, as we aren't checking against the clientID now yet.